107/src/main/java/org/ehcache/jsr107/ExpiryPolicyToEhcacheExpiry.java

```diff
@@ -55,28 +53,22 @@
       if (duration == null) {
         return null;
       }
-      if (duration.isEternal()) {
-        return org.ehcache.expiry.Duration.INFINITE;
-      }
-      return new org.ehcache.expiry.Duration(duration.getDurationAmount(), duration.getTimeUnit());
+      return convertDuration(duration);
     } catch (Throwable t) {
-      return org.ehcache.expiry.Duration.ZERO;
+      return java.time.Duration.ZERO;
     }
   }
 
   @Override
-  public org.ehcache.expiry.Duration getExpiryForUpdate(K key, ValueSupplier<? extends V> oldValue, V newValue) {
+  public java.time.Duration getExpiryForUpdate(K key, ValueSupplier<? extends V> oldValue, V newValue) {
     try {
       Duration duration = expiryPolicy.getExpiryForUpdate();
       if (duration == null) {
         return null;
       }
-      if (duration.isEternal()) {
-        return org.ehcache.expiry.Duration.INFINITE;
-      }
-      return new org.ehcache.expiry.Duration(duration.getDurationAmount(), duration.getTimeUnit());
+      return convertDuration(duration);
     } catch (Throwable t) {
-      return org.ehcache.expiry.Duration.ZERO;
+      return java.time.Duration.ZERO;
```

Comment:
Not related to your changes but isn't super weird to swallow the exception whatever it is?

Reply:
It is part of the contract, this behaviour is described in the `ExpiryPolicy` javadoc